js/src/forum/panes/ProfilePane.tsx

```diff
@@ -48,14 +57,17 @@ export default class ProfilePane extends Component {
 
   load() {
     this.answers = this.user.masqueradeAnswers();
+    const userId = this.user.id();
+
+    if (!userId) return;
 
-    if (this.answers === false) {
-      this.answers = [];
-      app.store.find('users', this.user.id(), { include: 'masqueradeAnswers' }).then(() => {
-        this.answers = this.user.masqueradeAnswers();
-        m.redraw();
-      });
-    }
+    if (this.answers) return;
+
+    this.answers = [];
+    app.store.find<User>('users', userId, { include: 'masqueradeAnswers' }).then(() => {
```

Comment:
What about converting this entire function to `async`?

Reply:
Yep, why not. Done in https://github.com/FriendsOfFlarum/masquerade/pull/89/commits/676a9b8fe6ed92f5c4b8ce73d2baa6e2b2579220